README.md
# [Sucharu Themes](https://hsbasu.github.io/mamolinux-themes)

<p align="center">
	<a href="https://github.com/hsbasu/mamolinux-themes/blob/master/LICENSE">
		<img src="https://img.shields.io/github/license/hsbasu/mamolinux-themes?label=License" alt="License"
	</a>
  	<a href="#">
		<img src="https://img.shields.io/github/repo-size/hsbasu/mamolinux-themes?label=Repo%20size" alt="GitHub repo size">
  	</a>
	<a href="https://github.com/hsbasu/mamolinux-themes/issues" target="_blank">
		<img src="https://img.shields.io/github/issues/hsbasu/mamolinux-themes?label=Issues" alt="Open Issues">
	</a>
	<a href="https://github.com/hsbasu/mamolinux-themes/pulls" target="_blank">
		<img src="https://img.shields.io/github/issues-pr/hsbasu/mamolinux-themes?label=PR" alt="Open PRs">
	</a>
  	<a href="https://github.com/hsbasu/mamolinux-themes/releases/latest">
		<img src="https://img.shields.io/github/v/release/hsbasu/mamolinux-themes?label=Latest%20Stable%20Release" alt="GitHub release (latest by date)">
  	</a>
	<a href="#download-latest-version">
		<img src="https://img.shields.io/github/downloads/hsbasu/mamolinux-themes/total?label=Downloads" alt="Downloads">
	</a>
	<a href="https://github.com/hsbasu/mamolinux-themes/releases/download/1.9.8mamolinux3/mamolinux-themes_1.9.8mamolinux3_all.deb">
		<img src="https://img.shields.io/github/downloads/hsbasu/mamolinux-themes/1.9.8mamolinux3/mamolinux-themes_1.9.8mamolinux3_all.deb?color=blue&label=Downloads%40Latest%20Binary" alt="GitHub release (latest by date and asset)">
	</a>
</p>

A GTK theme with mac-styled 3D title buttons.

## Download Latest Version
<p align="center">
	<a href="https://github.com/hsbasu/mamolinux-themes/zipball/master">Download Source (.zip)</a></br>
	<a href="https://github.com/hsbasu/mamolinux-themes/tarball/master">Download Source (.tar.gz)</a></br>
	<a href="https://github.com/hsbasu/mamolinux-themes/releases/download/1.9.8mamolinux3/mamolinux-themes_1.9.8mamolinux3_all.deb">Download Binary (.deb)</a>
</p>

## Features and Screenshots
          
### GTK theme
A GTK theme with mac-styled 3D title buttons.

This theme is based on [Mint-Y](https://github.com/linuxmint/mint-themes) themes.

<p align="center">
	<img src="#" alt="Main Window (Light)">
	<img src="#" alt="Main Window (Dark)">
</p>


## Contents
- [Download Latest Version](#download-latest-version)
- [Features and Screenshots](#features-and-screenshots)
- [Dependencies](#dependencies)
	- [Debian/Ubuntu based systems](#debianubuntu-based-distro)
	- [Other Linux-based systems](#other-linux-based-distro)
- [Installation](#build-and-install-the-latest-version)
	- [Debian/Ubuntu based systems](#debianubuntu-based-systems)
	- [Other Linux-based systems](#other-linux-based-systems)
	- [For Developers](#for-developers)
- [User Manual](#user-manual)
- [Issue Tracking and Contributing](#issue-tracking-and-contributing)
- [Contributors](#contributors)
	- [Authors](#author)

## Dependencies
```
debhelper ( > = 12)
gnome-themes-standard
gtk2-engines-murrine
gtk2-engines-pixbuf
inkscape
python3
sassc
```
### Debian/Ubuntu based distro
To install dependencies on Debian/Ubuntu based systems, run:
```
sudo apt install debhelper gnome-themes-standard gtk2-engines-murrine \
gtk2-engines-pixbuf inkscape python3 sassc
```

### Other Linux-based distro
Remove `apt install` in the command given in [Debian/Ubuntu based distros](#debianubuntu-based-distro) and use the command for the package manager of the target system(eg. `yum install`, `dnf install`, `pacman -S` etc.)

**Note**: There might be cases where one or more dependencies might not be available for your system. But that is highly unlikely. In such situations, please [create an issue](#issue-tracking-and-contributing).

## Build and Install the Latest Version
### Debian/Ubuntu based systems
There are two methods, these themes can be installed/used on a Debian/Ubuntu based system. First, download and unzip the source package using:
```
wget https://github.com/hsbasu/mamolinux-themes/archive/refs/heads/master.zip
unzip master.zip
cd mamolinux-themes-master
```

1. **Option 1:** Manually copying necessary files to root (`/`). For that, follow the steps below:
	1. To build the themes, run:
		```
		make clean
		make all
		```
		from the `/path/to/repo` in a terminal. It will create the **GTK** and **Cinnamon** themes in `usr/share/themes`, and **Plank themes** in `usr/share/plank/themes`.

	2. Copy the contents of `usr/` to `/usr/`:
		```
		sudo cp -R usr /
		```

2. **Option 2:** Build debian packages and install it. To build a debian package on your own:
	1. from the `/path/to/repo` run:
		```
		dpkg-buildpackage --no-sign
		```
		This will create `mamolinux-themes_*.deb` and `mamolinux-themes-plank_*.deb` package at `../path/to/repo`.
	2. Install the debian packages using
		```
		sudo dpkg -i *.deb
		sudo apt install -f
		```
	After it is installed, set the themes from your distro's theme manager or use the [Theme Manager](https://hsbasu.github.io/theme-manager).

### Other Linux-based systems
1. Install the [dependencies](#other-linux-based-distro).
2. From instructions for [Debian/Ubuntu based systems](#debianubuntu-based-systems), follow **Option 1**.

### For Developers
Coming Soon or create a PR.

**I have no knowledge on how to use `meson` or `npm` for testing. If you can offer any help regarding this, please start a discussion [here](https://github.com/hsbasu/mamolinux-themes/discussions) or create a [PR](https://github.com/hsbasu/mamolinux-themes/compare). It will be more than welcome.**

## User Manual
Coming Soon or create a PR.

## Issue Tracking and Contributing
If you are interested to contribute and enrich the code, you are most welcome. You can do it by:
1. If you find a bug, to open a new issue with details: [Click Here](https://github.com/hsbasu/mamolinux-themes/issues)
2. If you know how to fix a bug or want to add new feature/documentation to the existing package, please create a [Pull Request](https://github.com/hsbasu/mamolinux-themes/compare).

## Contributors

### Author
[Himadri Sekhar Basu](https://github.com/hsbasu) is the author and current maintainer.

## Donations
I am a freelance programmer. So, If you like this app and would like to offer me a coffee ( &#9749; ) to motivate me further, you can do so via:

[![](https://liberapay.com/assets/widgets/donate.svg)](https://liberapay.com/hsbasu/donate)
[![](https://www.paypalobjects.com/webstatic/i/logo/rebrand/ppcom.svg)](https://paypal.me/hsbasu)
[![](https://hsbasu.github.io/styles/icons/logo/svg/upi-logo.svg)](https://hsbasu.github.io/images/upi-qr.jpg)


debian/changelog
mint-themes (1.9.5) una; urgency=medium

  [ Michael Webster ]
  * Mint-Y: Fix top and bottom clipping of the window close button in client-decorated firefox.

  [ Clement Lefebvre ]
  * Mint-Y: Update metacity thumbnail

 -- Clement Lefebvre <root@linuxmint.com>  Fri, 17 Dec 2021 16:06:35 +0000

mint-themes (1.9.4) una; urgency=medium

  [ Michael Webster ]
  * Mint-X cinnamon: Add support for the no-events 'button' and improve the go-home button's appearance when enabled.
  * mint-x cinnamon: Some cleanup.
  * Mint-Y cinnamon: Add support for the 'no events' button.
  * Mint-X cinnamon: Fix color of no-events calendar icon.
  * Update the 'Linux Mint' cinnamon theme for calendar changes.

 -- Clement Lefebvre <root@linuxmint.com>  Thu, 16 Dec 2021 10:19:29 +0000

mamolinux-themes (1.9.3mamolinux1) stable; urgency=medium

  [ Himadri Sekhar Basu ]
  * Sucharu-Gtk3: Update assets svg
  * Sucharu: Update rendering scripts
  * Sucharu: Update png assets

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Sat, 11 Dec 2021 12:50:09 +0530

mint-themes (1.9.3) una; urgency=medium

  [ Michael Webster ]
  * Mint-X cinnamon: Fix hover style of 'home date' button, change color used for calendar day hover.

 -- Clement Lefebvre <root@linuxmint.com>  Thu, 09 Dec 2021 11:27:44 +0000

mamolinux-themes (1.9.2+mamolinux1) focal; urgency=medium

  * Follow the metacity rules from yaru-cinnamon
  * Don't track .css files as they are generated via scss
  * Mint-Y-Gtk3: Fix "-Darker" themes
  * Gtk2: Update "-Darker" theme
  * Use multiple processors to generate assets faster
  * Rename theme to Sucharu
  * Sucharu-cinnamon: fix panel colors
  * Fix generate and test scripts
  * Fix dependencies and update homepage

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Thu, 09 Dec 2021 16:55:17 +0530

mint-themes (1.9.2) una; urgency=medium

  * Xfwm4: titlebar color fixes
  * Xfwm4: Make bottom corners completely square
  * Generate asssets

 -- Clement Lefebvre <root@linuxmint.com>  Fri, 03 Dec 2021 15:31:31 +0000

mint-themes (1.9.1) una; urgency=medium

  [ Michael Webster ]
  * Add additional calendar/event support to Cinnamon themes.

  [ Clement Lefebvre ]
  * xfwm4: Don't hardcode the button layout
  * Xfwm4: Use circle for shade and rivet for menu
  * Update assets

 -- Clement Lefebvre <root@linuxmint.com>  Fri, 03 Dec 2021 14:35:49 +0000

mint-themes (1.9.0) una; urgency=medium

  [ Michael Webster ]
  * Mint-Y-Cinnamon: Don't allow the number of calendar events to affect the popup's height.

  [ Clement Lefebvre ]
  * Gtk2: Use generated assets
  * Remove tint on all variations
  * Regenerate assets
  * Update Xfwm4 theme
  * Update variations

 -- Clement Lefebvre <root@linuxmint.com>  Fri, 03 Dec 2021 11:48:12 +0000

mamolinux-themes (1.8.9) una; urgency=medium

  [ JosephMcc ]
  * Mint-Y-Gtk2: Put separators in menus
  * Gtk3: Better match csd and ssd window shadows
  * Mint-X-Gtk3: Add styling for app-notification
  * test-mint-y: Add a couple more things
  * Mint-Y: Remove the -darker variation
  * Mint-Y: Add a dark variant to the light theme
  * Mint-Y: Only ship a single metacity theme
  * Mint-Y: Force the metacity theme to update on gtk theme changes
  * Mint-Y-Cinnamon: Set a height for the run dialog entry
  * Mint-Y: Use a light sidebar in file managers
  * Mint-Y-Cinnamon: Give the light theme a light panel
  * Mint-Y: Restyle the window decorations
  * Mint-Y: Missed including a new icon for metacity
  * Mint-Y-Gtk3: Fix osd colors
  * Mint-Y: Tint the entire theme with the selected_bg_color
  * Mint-Y: Add rounded corners to the window borders
  * Mint-Y: Adjust csd window shadows again
  * Mint-Y-Gtk3: Fix the hover color in the Nemo sidebar
  * Mint-Y: Add a light 1px border around window frames
  * Mint-Y: Bump up the color mix a bit
  * Mint-Y: Match up csd and non-csd window borders a bit better
  * Mint-Y: Hide shadows on tiled csd windows
  * Mint-Y: Improve the styling of the Caja filemanager
  * Mint-Y: Update the styling of the Thunar sidebar
  * Mint-Y: Fix the color of the Mate panel
  * Mint-Y: Clean up the titlebar styling some more
  * Mint-Y: Fix the size of csd titlebuttons
  * Mint-Y-Cinnamon: Adapt to the Cinnamon modal dialog theming changes
  * Mint-Y: Don't use the accent color in active buttons
  * Mint-Y: Fix the color of focus outline on active buttons
  * Mint-Y: Don't use the accent color for active menuitems

  [ Clement Lefebvre ]
  * Fix test script
  * Mint-Y: Metacity, fix artefacts in close button.

  [ JosephMcc ]
  * Mint-Y: Don't special case sidebar scrollbars
  * Mint-Y: Readapt the csd titlebuttons to match metacity
  * Mint-Y: Fix corner radius of ssd window decorations

  [ Clement Lefebvre ]
  * Remove tint on green theme and base color

  [ JosephMcc ]
  * Mint-Y-Dark: Fix the label color of events in gnome calendar
  * Mint-Y-Cinnamon: Add initial support for the reworked calendar applet

 -- Clement Lefebvre <root@linuxmint.com>  Fri, 26 Nov 2021 14:29:02 +0000

mamolinux-themes (1.8.9experimental+freedom1) stable; urgency=medium

  [ JosephMcc ]
  * Mint-Y: Force the metacity theme to update on gtk theme changes
  * Mint-Y-Cinnamon: Set a height for the run dialog entry
  * Mint-Y: Use a light sidebar in file managers
  * Mint-Y-Cinnamon: Give the light theme a light panel
  * Mint-Y: Restyle the window decorations
  * Mint-Y: Missed including a new icon for metacity
  * Mint-Y-Gtk3: Fix osd colors
  
  [ Himadri Sekhar Basu ]
  * Mint-Y-Gtk3: Introduce light and dark mode for terminal

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Wed, 24 Nov 2021 09:58:13 +0530

mint-themes (1.8.8+freedom1) uma; urgency=medium

  [ Himadri Sekhar Basu ]
  * Remove Mint-X and Linux Mint themes
  * Modify panel colours.
  * Modify variation "Teal"
  * Modify variation "Sand"
  * Modify variation "Red"
  * Modify variation "Purple"
  * Modify variation "Pink"
  * Modify variation "Orange"
  * Modify variation "Grey"
  * Modify variation "Brown"

constants.py
#!/usr/bin/python3

# Mint-Y
Y_HEX_ACCENT1 = ["#92b372"]  # BASE
Y_HEX_ACCENT2 = ["#8fa876"]  # DARK
Y_HEX_ACCENT3 = ["#afca95"]  # PRELIGHT/HOVER
Y_HEX_ACCENT4 = ["#779559"]  # PRESSED

y_hex_colors1 = {}
y_hex_colors2 = {}
y_hex_colors3 = {}
y_hex_colors4 = {}

y_hex_colors1["Aqua"] = "#6cabcd"
y_hex_colors2["Aqua"] = "#6aa0bd"
y_hex_colors3["Aqua"] = "#82b3ce"
y_hex_colors4["Aqua"] = "#52819a"

y_hex_colors1["Blue"] = "#5b73c4"
y_hex_colors2["Blue"] = "#596eb5"
y_hex_colors3["Blue"] = "#6e82c6"
y_hex_colors4["Blue"] = "#4b5a8d"

y_hex_colors1["Brown"] = "#aa876a"
y_hex_colors2["Brown"] = "#9c7e65"
y_hex_colors3["Brown"] = "#ad9078"
y_hex_colors4["Brown"] = "#796555"

y_hex_colors1["Grey"] = "#9d9d9d"
y_hex_colors2["Grey"] = "#8f8f8f"
y_hex_colors3["Grey"] = "#a3a3a3"
y_hex_colors4["Grey"] = "#727272"

y_hex_colors1["Orange"] = "#db9d61"
y_hex_colors2["Orange"] = "#cc9560"
y_hex_colors3["Orange"] = "#dcaa7a"
y_hex_colors4["Orange"] = "#a97747"

y_hex_colors1["Pink"] = "#c76199"
y_hex_colors2["Pink"] = "#b85f90"
y_hex_colors3["Pink"] = "#c975a3"
y_hex_colors4["Pink"] = "#924d73"

y_hex_colors1["Purple"] = "#8c6ec9"
y_hex_colors2["Purple"] = "#866cba"
y_hex_colors3["Purple"] = "#9b84cb"
y_hex_colors4["Purple"] = "#6a5497"

y_hex_colors1["Red"] = "#c15b58"
y_hex_colors2["Red"] = "#b35a57"
y_hex_colors3["Red"] = "#d1716f"
y_hex_colors4["Red"] = "#9a4a47"

y_hex_colors1["Sand"] = "#c8ac69"
y_hex_colors2["Sand"] = "#b89f65"
y_hex_colors3["Sand"] = "#c9b27c"
y_hex_colors4["Sand"] = "#947f51"

y_hex_colors1["Teal"] = "#5aaa9a"
y_hex_colors2["Teal"] = "#579c8e"
y_hex_colors3["Teal"] = "#6dbbab"
y_hex_colors4["Teal"] = "#4e8378"

y_hex_colors1["Yellow"] = "#e6e600"
y_hex_colors2["Yellow"] = "#d9d90b"
y_hex_colors3["Yellow"] = "#fffacd"
y_hex_colors4["Yellow"] = "#e6c200"


update-variations.py
#!/usr/bin/python3
import os
import sys

from constants import Y_HEX_ACCENT1, Y_HEX_ACCENT2, Y_HEX_ACCENT3, Y_HEX_ACCENT4
from constants import y_hex_colors1, y_hex_colors2, y_hex_colors3, y_hex_colors4

def change_value (key, value, file):
    if value is not None:
        command = "sed -i '/%(key)s=/c\%(key)s=%(value)s' %(file)s" % {'key':key, 'value':value, 'file':file}
    else:
        command = "sed -i '/%(key)s=/d' %(file)s" % {'key':key, 'file':file}
    os.system(command)

def usage ():
    print ("Usage: update-variations.py color")
    print ("color can be 'Aqua', 'Blue', 'Brown', 'Grey', 'Orange', 'Pink', 'Purple', 'Red', 'Sand', 'Teal', 'Yellow' or 'All'.")
    sys.exit(1)

def update_color (color):
    variation = curdir+"/src/Mint-Y/variations/%s" % color
    rendering_script = curdir+"/src/Mint-Y/render-assets.sh"
    print("updating %s" % color)
    os.system("rm -rf %s" % variation)
    os.system("mkdir -p %s/gtk-2.0" % variation)
    os.system("mkdir -p %s/gtk-3.0" % variation)
    os.system("mkdir -p %s/gtk-4.0" % variation)
    os.system("mkdir -p %s/xfwm4" % variation)
    os.system("mkdir -p %s/xfwm4-dark" % variation)

    # Copy assets files
    assets = []
    assets.append("gtk-2.0/assets.svg")
    assets.append("gtk-2.0/assets-dark.svg")
    assets.append("gtk-3.0/assets.svg")
    assets.append("gtk-4.0/assets.svg")
    assets.append("xfwm4/assets.svg")
    assets.append("xfwm4-dark/assets.svg")

    files = []
    files.append("gtk-2.0/assets")
    files.append("gtk-2.0/assets-dark")
    files.append("gtk-2.0/assets.txt")
    files.append("gtk-3.0/assets")
    files.append("gtk-3.0/assets.txt")
    files.append("gtk-4.0/assets")
    files.append("gtk-4.0/assets.txt")
    files.append("xfwm4/assets.txt")
    files.append("xfwm4-dark/assets.txt")

    for file in files:
        os.system("cp -R src/Mint-Y/%s %s/%s" % (file, variation, file))
    for asset in assets:
        os.system("cp -R src/Mint-Y/%s %s/%s" % (asset, variation, asset))

    # Update assets svg
    for asset in assets:
        asset_path = "%s/%s" % (variation, asset)
        for accent in Y_HEX_ACCENT1:
            os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors1[color], 'file': asset_path})
        for accent in Y_HEX_ACCENT2:
            os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors2[color], 'file': asset_path})
        for accent in Y_HEX_ACCENT3:
            os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors3[color], 'file': asset_path})
        for accent in Y_HEX_ACCENT4:
            os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors4[color], 'file': asset_path})

    # Render assets
    # TODO: need better idea to do '-dark'
    # and '@2' arguments
    os.chdir(variation)
    os.chdir(variation+"/gtk-2.0")
    print("**Rendering gtk-2.0 assets...")
    os.system("rm -rf assets/*")
    os.system(rendering_script)
    
    print("**Rendering gtk-2.0 dark assets...")
    os.system("rm -rf assets-dark/*")
    os.system(rendering_script+" -dark")
    # os.system("rm -rf assets/*@2.png")
    
    print("**Rendering gtk-3.0 assets...")
    os.chdir(variation+"/gtk-3.0/")
    os.system("rm -rf assets/*")
    os.system(rendering_script+" s2")
    
    print("**Rendering gtk-4.0 assets...")
    os.chdir(variation+"/gtk-4.0/")
    os.system("rm -rf assets/*")
    os.system(rendering_script+" s2")
    
    print("**Rendering xfwm4 assets...")
    os.chdir(variation+"/xfwm4/")
    os.system("rm -rf *.png")
    os.system(rendering_script)
    # os.system("rm -rf assets/*@2.png")
    os.system("mv assets/*.png ./ && rm -rf assets/")
    
    print("**Rendering xfwm4 dark assets...")
    os.chdir(variation+"/xfwm4-dark/")
    os.system("rm -rf *.png")
    os.system(rendering_script)
    # os.system("rm -rf assets/*@2.png")
    os.system("mv assets/*.png ./ && rm -rf assets/")
    print("")
    os.chdir(curdir)

if len(sys.argv) < 2:
    usage()
else:
    color_variation = sys.argv[1]
    if not color_variation in ["Aqua", "Blue", "Brown", "Grey", "Orange", "Pink", "Purple", "Red", "Sand", "Teal", "Yellow", "All"]:
        usage()

# Mint-Y variations
curdir = os.getcwd()

if color_variation == "All":
    for color in y_hex_colors1.keys():
        update_color(color)
else:
    update_color(color_variation)


generate-themes.py
#!/usr/bin/python3
import os, sys

from constants import Y_HEX_ACCENT1, Y_HEX_ACCENT2, Y_HEX_ACCENT3, Y_HEX_ACCENT4
from constants import y_hex_colors1, y_hex_colors2, y_hex_colors3, y_hex_colors4

def change_value (key, value, file):
    if value is not None:
        command = "sed -i '/%(key)s=/c\%(key)s=%(value)s' %(file)s" % {'key':key, 'value':value, 'file':file}
    else:
        command = "sed -i '/%(key)s=/d' %(file)s" % {'key':key, 'file':file}
    os.system(command)
    
def usage ():
    print ("Usage: generate-themes.py [color]")
    print ("color can be 'Aqua', 'Blue', 'Brown', 'Grey', 'Orange', 'Pink', 'Purple', 'Red', 'Sand', 'Teal' or 'All'.")
    sys.exit(1)

def y_colorize_directory (path, variation):
    for accent in Y_HEX_ACCENT1:
        os.system("find %s -name '*.*' -type f -exec sed -i 's/%s/%s/gI' {}  \\;" % (path, accent, y_hex_colors1[variation]))
    for accent in Y_HEX_ACCENT2:
        os.system("find %s -name '*.*' -type f -exec sed -i 's/%s/%s/gI' {}  \\;" % (path, accent, y_hex_colors2[variation]))
    for accent in Y_HEX_ACCENT3:
        os.system("find %s -name '*.*' -type f -exec sed -i 's/%s/%s/gI' {}  \\;" % (path, accent, y_hex_colors3[variation]))
    for accent in Y_HEX_ACCENT4:
        os.system("find %s -name '*.*' -type f -exec sed -i 's/%s/%s/gI' {}  \\;" % (path, accent, y_hex_colors4[variation]))

def generate_theme(color):
    # build Sucharu color variations
    # for color in y_hex_colors1.keys():
    for variant in ["", "-Dark", "-Darker"]:
        original_name = "Sucharu%s" % variant
        path = os.path.join("src/Mint-Y/variations/%s" % color)
        if os.path.isdir(path):
            print("Derivating %s-%s" % (original_name, color))
            
            # Copy theme
            theme = "usr/share/themes/%s-%s" % (original_name, color)
            theme_index = os.path.join(theme, "index.theme")
            os.system("cp -R usr/share/themes/%s %s" % (original_name, theme))
            
            # Theme name
            for key in ["Name", "GtkTheme"]:
                change_value(key, "%s-%s" % (original_name, color), theme_index)
            
            for key in ["MetacityTheme"]:
                metacity_variant = original_name.replace("Darker", "Dark")
                change_value(key, "%s-%s" % (metacity_variant, color), theme_index)
            
            for key in ["IconTheme"]:
                change_value(key, "%s-%s" % (original_name, color), theme_index)
            
            for key in ["CursorTheme"]:
                change_value(key, "FreedomOS-%s" % color, theme_index)
            
            # Regenerate GTK4 sass
            os.system("cp -R src/Mint-Y/gtk-4.0/sass %s/gtk-4.0/" % theme)
            y_colorize_directory("%s/gtk-4.0/sass" % theme, color)
            os.chdir("%s/gtk-4.0" % theme)

            if (variant == "-Dark"):
                os.system("cp sass/gtk-dark.scss sass/gtk.scss")
                os.system("sassc ./sass/gtk.scss gtk.css")
            else:
                os.system("sassc ./sass/gtk-dark.scss gtk-dark.css")
                os.system("sassc ./sass/gtk.scss gtk.css")

            os.system("rm -rf sass .sass-cache")
            os.chdir(curdir)

            # Regenerate GTK3 sass
            os.system("cp -R src/Mint-Y/gtk-3.0/sass %s/gtk-3.0/" % theme)
            y_colorize_directory("%s/gtk-3.0/sass" % theme, color)
            os.chdir("%s/gtk-3.0" % theme)
            # os.system("sed -i 's/no-tint/tint/gI' ./sass/gtk.scss")
            # os.system("sed -i 's/no-tint/tint/gI' ./sass/gtk-dark.scss")
            if (variant == "-Dark"):
                os.system("cp sass/gtk-dark.scss sass/gtk.scss")
            elif (variant == "-Darker"):
                os.system("cp sass/gtk-darker.scss sass/gtk.scss")
            else:
                os.system("rm sass/gtk-dark.scss sass/gtk-darker.scss")
            
            os.system("sassc ./sass/gtk.scss gtk.css")
            os.system("rm -rf sass .sass-cache")
            os.chdir(curdir)
            
            # Regenerate Cinnamon sass
            if (variant != "-Darker"):
                # Darker variants have no cinnamon style
                os.system("cp -R src/Mint-Y/cinnamon/sass %s/cinnamon/" % theme)
                y_colorize_directory("%s/cinnamon/sass" % theme, color)
                os.chdir("%s/cinnamon" % theme)
                if (variant == "-Dark"):
                    os.system("cp sass/cinnamon-dark.scss sass/cinnamon.scss")
                os.system("sassc ./sass/cinnamon.scss cinnamon.css")
                os.system("rm -rf sass .sass-cache")
            os.chdir(curdir)
            
            # Accent color
            files = []
            files.append(os.path.join(theme, "gtk-2.0", "gtkrc"))
            files.append(os.path.join(theme, "gtk-2.0", "main.rc"))
            files.append(os.path.join(theme, "gtk-2.0", "panel.rc"))
            files.append(os.path.join(theme, "gtk-2.0", "apps.rc"))
            files.append(os.path.join(theme, "gtk-2.0", "menubar-toolbar.rc"))
            for file in files:
                if os.path.exists(file):
                    for accent in Y_HEX_ACCENT1:
                        os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors1[color], 'file': file})
                    for accent in Y_HEX_ACCENT2:
                        os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors2[color], 'file': file})
                    for accent in Y_HEX_ACCENT3:
                        os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors3[color], 'file': file})
                    for accent in Y_HEX_ACCENT4:
                        os.system("sed -i s'/%(accent)s/%(color_accent)s/gI' %(file)s" % {'accent': accent, 'color_accent': y_hex_colors4[color], 'file': file})
            
            # Remove metacity-theme-3.xml (it doesn't need to be derived since it's using GTK colors, and Cinnamon doesn't want to list it)
            os.system("rm -rf %s" % os.path.join(theme, "metacity-1"))
            
            directories = []
            directories.append(os.path.join(theme, "cinnamon/common-assets"))
            directories.append(os.path.join(theme, "cinnamon/light-assets"))
            directories.append(os.path.join(theme, "cinnamon/dark-assets"))
            for directory in directories:
                if os.path.exists(directory):
                    y_colorize_directory(directory, color)
            
            # Assets
            os.system("rm -rf %s/gtk-4.0/assets" % theme)
            os.system("rm -rf %s/gtk-3.0/assets" % theme)
            os.system("rm -rf %s/gtk-2.0/assets" % theme)
            if variant == "-Dark":
                os.system("cp -R %s/gtk-2.0/assets-dark %s/gtk-2.0/assets" % (path, theme))
                os.system("cp -R %s/xfwm4-dark/*.png %s/xfwm4/" % (path, theme))
            else:
                os.system("cp -R %s/gtk-2.0/assets %s/gtk-2.0/assets" % (path, theme))
                os.system("cp -R %s/xfwm4/*.png %s/xfwm4/" % (path, theme))
            os.system("cp -R %s/gtk-3.0/assets %s/gtk-3.0/assets" % (path, theme))
            os.system("cp -R %s/gtk-4.0/assets %s/gtk-4.0/assets" % (path, theme))
            os.system("cp -R files/%s ./usr/share/themes" % theme)


if len(sys.argv) < 2:
    usage()
else:
    color_variation = sys.argv[1]
    if not color_variation in ["Aqua", "Blue", "Brown", "Grey", "Orange", "Pink", "Purple", "Red", "Sand", "Teal", "Yellow", "All"]:
        usage()

curdir = os.getcwd()

# if os.path.exists("usr"):
#         os.system("rm -rf usr/")

os.system("mkdir -p usr/share/themes")

# Build Sucharu Base themes (light, dark, darker)
os.chdir("src/Mint-Y")
os.system("./build-themes.py")
os.chdir(curdir)

if color_variation == "All":
    for color in y_hex_colors1.keys():
        generate_theme(color)
else:
    generate_theme(color_variation)
    